### PR TITLE
CI: Add build docs

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

**Comment by whatyouhide** [comment]
Is the idea with this to build docs just to ensure that there are no errors when building docs?

**Comment by josevalim** [comment]
@whatyouhide yes, because it was broken due to a couple ExDocs commits prior. We can also add --warnings-as-errors in the future to handle invalid docs. :) Does it look good to you?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -52,7 +56,16 @@ jobs:
...
         run: taskset 1 make check_reproducible
-        if: matrix.otp_release == 'OTP-24.0'
+        if: ${{ matrix.otp_latest }}
+      - name: Build docs
+        if: ${{ matrix.otp_latest }}
+        run: |
+          cd ..
+          git clone https://github.com/elixir-lang/ex_doc.git --branch main --depth 1
+          cd ex_doc
+          ../elixir/bin/mix do local.rebar --force, local.hex --force, deps.get, compile
```
Comment (josevalim) [inline .github/workflows/ci.yml:66]:
```suggestion
          ../elixir/bin/mix do local.rebar --force + local.hex --force + deps.get + compile
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by eksperimental** [comment]
> @whatyouhide yes, because it was broken due to a couple ExDocs commits prior. We can also add --warnings-as-errors in the future to handle invalid docs. :) Does it look good to you?

@whatyouhide this is the issue in ExDoc for reference https://github.com/elixir-lang/ex_doc/issues/1539

File: .github/workflows/ci.yml (hunk 1 above)
Comment (eksperimental) [reply .github/workflows/ci.yml]: @josevalim I know this has been recently introduced in Elixir. So are we deprecating the usage of comma as separator?

Comment (josevalim) [reply .github/workflows/ci.yml]: In a far away future yes.

Comment (eksperimental) [reply .github/workflows/ci.yml]: Sorry, what I meant to ask was, are we replacing commas with plus symbols in Elixir?

Comment (josevalim) [reply .github/workflows/ci.yml]: No. This is unrelated to the language syntax. :)

Comment (eksperimental) [reply .github/workflows/ci.yml]: Sorry, this is what I mean.

```
$ ag "mix\s+do\s+[^\n]+, " -U --ignore doc
README.md
192:cd ex_doc && ../elixir/bin/mix do deps.get, compile

lib/mix/lib/mix/tasks/do.ex
39:      mix do compile --list, deps

lib/mix/lib/mix/tasks/archive.install.ex
13:      mix do archive.build, archive.install

lib/mix/lib/mix/tasks/escript.install.ex
13:      mix do escript.build, escript.install

lib/mix/lib/mix/tasks/test.ex
187:          MIX_ENV=test mix do compile --warnings-as-errors, test --warnings-as-errors

```

should these matches be replaced with `+` ?

Comment (josevalim) [reply .github/workflows/ci.yml]: Yes!

**Pull request merged**